substrate/network/src/protocol.rs

```diff
@@ -511,6 +513,8 @@ impl Protocol {
 				}));
 			}
 		}
+
+		self.consensus.lock().collect_garbage(Some(header.parent_hash));
```

Comment:
if this doesn't fire every block then we will have a lot of garbage piling up.

Reply:
seems better to keep track of the header hash of the last notification and pass all the parent headers since then.